src/App.tsx

```diff
@@ -15,71 +20,89 @@ export const goods = [
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        { this.state.selectedGood
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${this.state.selectedGood} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.setState({ selectedGood: '' })}
+              />
+            </h1>
+          )
+          : (
+            <h1 className="title">
+              No goods selected
+            </h1>
+          )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isChosen = selectedGood === good;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <>
+                  <tr
+                    data-cy="Good"
+                    className={classNames(
+                      {
+                        'has-background-success-light': isChosen,
+                      },
+                    )}
+                  >
+                    <td>
+                      <button
+                        data-cy={classNames(
+                          {
+                            AddButton: !isChosen,
+                            RemoveButton: isChosen,
+                          },
+                        )}
+                        type="button"
+                        className={classNames('button',
+                          {
+                            'is-info': isChosen,
+                          })}
+                        onClick={() => {
```

Comment:
Move this to the separate function
```suggestion
                        onClick={handleSelect
```

Reply:
I cant going that because I need the variable isSelected, and if I try to pass this variable to the method, then I will have a call immediately to the function in place.
![image](https://user-images.githubusercontent.com/106397016/233109717-9cde36c2-2ff6-462f-842c-aa74daf5f463.png)

Reply:
Ofc, bc callback of onClick try to pass event to your fucntion
so, you need to do like this
```js
  onClick={(() => this.handleSelect(todo))};
```